### PR TITLE
Feature/process adjudication data

### PR DESCRIPTION
**Pull request by laurejt**
Ref: #84 

Script for processing prodigy adjudication data files (.jsonl). Also adds project support for type checking via mypy

### CODE REVIEW

File: src/corppa/poetry_detection/annotation/process_adjudication_data.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
+import argparse
```
Comment (rlskoeser) [inline src/corppa/poetry_detection/annotation/process_adjudication_data.py:1]: Please add a short docstring documenting briefly what this script does, what it is for, and one example of how to run it.

Here's how I ran it locally on the round 2 adjudication data:
```sh
python src/corppa/poetry_detection/annotation/process_adjudication_data.py allround2.jsonl round2_pages round2_excerpts
```

It wasn't obvious to me from the help information that I should have probably named the second param `round2_pages.jsonl` and the second one `round2_excerpts.csv`. I wasn't sure why we need both outputs or what the goals are for the different outputs.

File: src/corppa/poetry_detection/annotation/process_adjudication_data.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+    for excerpt in page_data["excerpts"]:
+        entry = {
+            "page_id": page_data["page_id"],
+            "work_id": page_data["work_id"],
+            "work_title": page_data["work_title"],
+            "work_author": page_data["work_author"],
+            "work_year": page_data["work_year"],
+            "start": excerpt["start"],
+            "end": excerpt["end"],
+            "text": excerpt["text"],
```
Comment (rlskoeser) [inline src/corppa/poetry_detection/annotation/process_adjudication_data.py:76]: when/if we come back to this (if we use this script again), let's tweak these fields to match the ones we've agreed on for the poem dataset

**Review by rlskoeser — APPROVED** [review]
Looks good, was able to run it on the round2 adjudication data and review the results.  Just needs a bit of documentation on what this is and how to run it, then it can be merged.

File: src/corppa/poetry_detection/annotation/process_adjudication_data.py (hunk 1 above)
Comment (laurejt) [reply src/corppa/poetry_detection/annotation/process_adjudication_data.py:1]: Just want to verify that the parser help descriptions are clear. For example, `output_pages` was described as "Filename where extracted pages data (JSONL file) should be written".

Comment (rlskoeser) [reply src/corppa/poetry_detection/annotation/process_adjudication_data.py:1]: "extracted pages data" was ambiguous to me in this context, I didn't know what it included or how it differed from the other output until poking around in the files a bit.

Maybe something like annotations aggregated/grouped by page  ?

Comment (laurejt) [reply src/corppa/poetry_detection/annotation/process_adjudication_data.py:23]: Does it make more sense in context now?

Comment (laurejt) [reply src/corppa/poetry_detection/annotation/process_adjudication_data.py:23]: I'm fine with changing the name later, but not something worth burning more of my R&D time on.

Comment (rlskoeser) [reply src/corppa/poetry_detection/annotation/process_adjudication_data.py:23]: thanks for updating the help text, it is clearer now

**Pull request merged**